### DISCUSSION TITLE
How to get the name of the input in the validation function

### DISCUSSION THREAD

**Discussion by christopher-francisco** [react-hook-form/react-hook-form#2584] [Q&A]
We have a list of checkbox inputs with a name that follows the array with string indexes:
```
"name"="days[monday]"
"name"="days[tuesday]"
"name"="days[wednesday]"
```

So the data comes back as a key-value dictionary:

```json
"days": {
  "monday": true,
  "tuesday": false,
  "wednesday": true
}
```

In the validation function, the `value` argument is of course a boolean, but I don't know how to get a reference of the particular checkbox:

```javascript
const validateCheckbox = value => {
  // The value here comes as `true`. However I don't know what `day` am I validating.
  // I'm missing the input `name` or the index, so that I know I'm validating the monday one

   // this gets me all the values, but I still don't know which day am I currently validating
   const { days } = getValues()
}

return (
  <>
    {["monday", "tuesday", "wednesday"].map(day => (
       <input type="checkbox" name={`days[${day}]`} ref={register({ validate: validateCheckbox })} />
    ))}
  </>
)
```

It feels like the input `name` is missing in the validate function

**Comment by HadiSDev** [comment]
`@chris-fran  Use custom register ref={(props: React.CheckboxHTMLAttributes<any>) => register({... something with **props.name** })}`

**Reply by christopher-francisco** [reply]
Not sure I follow, could you elaborate?

**Comment by christopher-francisco** [comment]
@bluebill1049 Sorry to ping you directly, have searched for docs but no luck

**Reply by bluebill1049** [reply]
you can get the name but then you will need callback wrap around register

`(ref) => register()`
